src/main/java/uk/gov/digital/ho/hocs/casework/api/StageService.java

```diff
@@ -141,21 +151,113 @@ StageWithCaseData getActiveStage(UUID caseUUID, UUID stageUUID) {
     }
 
     @Transactional
-    Stage createStage(UUID caseUUID, String stageType, UUID teamUUID, UUID userUUID, String emailType, UUID transitionNoteUUID) {
-        log.debug("Creating Stage of type: {}", stageType);
-        Stage stage = new Stage(caseUUID, stageType, teamUUID, userUUID, transitionNoteUUID);
+    public Stage createStage(UUID caseUUID, CreateStageRequest createStageRequest) {
+
+        if (caseUUID == null || createStageRequest == null) {
+            String msg = String.format("Missing required info to create new stage; caseUUID: %s, createStageRequest: %s", caseUUID, createStageRequest);
+            log.error(msg);
+            throw new ApplicationExceptions.EntityCreationException(msg, LogEvent.STAGE_CREATE_FAILURE);
+        }
+
+        log.info("Creating Stage: {} for case {}", createStageRequest, caseUUID);
+
+        // find current active stage to deactivate
+        StageWithCaseData activeStageToBeDeactivated = getCurrentActiveStageWithCaseData(caseUUID);
```

Comment:
I was concerned about the number of calls in this one, however the use of them was largely due to downstream processes requiring case data info, and thought it better to make this, rather than a call to get the case data as well. I think the required piece of case data was the case reference for notifications I believe.  I'll double check and maybe check into whether there is/or just create a call for that info only.

Reply:
I've not checked 100% but I think there were two calls, one that got the basic stage and one that got the basic case object but those two calls combined are still 500x quicker than this call 😀

Reply:
I've made changes to this to call only the Stage and separately call CaseData when needed.